mods/persistence/controllers/subsystems/initialization/chargen.dm

```diff
@@ -28,16 +27,99 @@ SUBSYSTEM_DEF(chargen)
 				CHECK_TICK
 
 /datum/controller/subsystem/chargen/proc/get_spawn_turf()
-	return pick(chargen_spawns)
+	//Grab first free pod
+	for(var/area/chargen/A in chargen_areas)
+		if(chargen_areas[A] > 0)
+			continue
+		var/obj/abstract/landmark/chargen_spawn/L = locate(/obj/abstract/landmark/chargen_spawn) in A
+		var/turf/spawnturf = L? get_turf(L) : null
+		if(!L)
+			log_warning("Area '[log_info_line(A)]' contains no '/obj/abstract/landmark/chargen_spawn' for spawning characters..")
+			spawnturf = locate(/turf/unsimulated/floor) in A
+		return spawnturf
+
+	var/chargen_warning = "SSChargen: Warning, the system is currently out of spawn pods to spawn new players in!"
+	message_staff(chargen_warning)
+	log_warning(chargen_warning)
+	return null
+
+/datum/controller/subsystem/chargen/proc/assign_spawn_pod(var/area/chargen/pod)
+	chargen_areas[pod] += 1
+
+	//Remove it from the list of landmarks
+	if(!pod.chargen_landmark)
+		log_warning("Chargen pod '[pod]' has not landmark set!")
+		return
+	
+	//Because spawnpoints don't have a proc to return turfs and instead just share their turf var to grab the available turfs, we gotta change the spawnpoint :D
+	var/decl/spawnpoint/chargen/C = GET_DECL(/decl/spawnpoint/chargen)
+	if(C)
+		LAZYREMOVE(C.turfs, get_turf(pod.chargen_landmark))
+	log_debug("SSChargen: Assigned area '[pod]' with '[chargen_areas[pod]]' assigned users currently!")
+
+/datum/controller/subsystem/chargen/proc/release_spawn_pod(var/area/chargen/pod)
+	chargen_areas[pod] -= 1
+
+	//Add it to the list of landmarks
+	if(!pod.chargen_landmark)
+		log_warning("Chargen pod '[pod]' has not landmark set!")
+		return
+
+	//Because spawnpoints don't have a proc to return turfs and instead just share their turf var to grab the available turfs, we gotta change the spawnpoint :D
+	var/decl/spawnpoint/chargen/C = GET_DECL(/decl/spawnpoint/chargen)
+	if(C)
+		LAZYDISTINCTADD(C.turfs, get_turf(pod.chargen_landmark))
 
-/datum/controller/subsystem/chargen/proc/assign_spawn_pod(var/mob/user, var/turf/location)
-	var/area/chargen/A = get_area(location)
-	A.assigned_to = user
-	chargen_spawns -= location
+	//Remove trash from the room
+	pod.run_chargen_cleanup()
+	log_debug("SSChargen: Un-assaigned area '[pod]' with '[chargen_areas[pod]]' assigned users currently!")
 
+//Landmark
 /obj/abstract/landmark/chargen_spawn
-	delete_me = TRUE
+	delete_me = FALSE //Currently re-used to keep track of the spawn position
 
 /obj/abstract/landmark/chargen_spawn/Initialize()
-	chargen_spawns += loc
+	var/area/chargen/A = get_area(src)
+	if(!istype(A))
+		log_warning("[src] is outside of a '/area/chargen' area!! Only place '/obj/abstract/landmark/chargen_spawn' inside a '/area/chargen'!!")
```

Comment:
Extra ! at the end

Reply:
Its very important, so it gets two exclamation marks! D: